packages/grid/src/GridMetricCalculator.ts

```diff
@@ -312,8 +323,9 @@ class GridMetricCalculator {
           )
         : 0;
 
-    const leftColumnWidth = visibleColumnWidths.get(left) ?? 0;
-    const topRowHeight = visibleRowHeights.get(top) ?? 0;
+    const leftColumnWidth =
+      columnCount > 0 ? mapGet(visibleColumnWidths, left) : 0;
+    const topRowHeight = rowCount > 0 ? mapGet(visibleRowHeights, top) : 0;
```

Comment:
`mapGet` makes TS more opaque in this case. If someone were to write something similar but forget to check the size, then `mapGet` would throw, but only when it's in use as opposed to the static compile time error of the regular `get` possibly being undefined

Reply:
Yes, we want mapGet to fail fast when used incorrectly.